libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerAction.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Adaptive;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Events;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.QnA
+{
+    /// <summary>
+    /// QnAMaker dialog which uses QnAMaker to get an answer.
+    /// </summary>
+    /// <remarks>
+    /// The answer is treated as a inline LG expression, so it can be completely customized by LG system.
+    /// </remarks>
+    public class QnAMakerAction : Dialog
+    {
+        private QnAMaker qnamaker;
+        private readonly HttpClient httpClient;
+
+        public QnAMakerAction(string kbId, string hostName, string endpointKey, HttpClient httpClient = null, [CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
```

Comment:
This ctor seems more complex thanneeded. Can it be redone to be simpler? 

Also, use the HttpClientFactory and related types. Do not pass around the HttpClient.